File: plugin/src/main/java/com/stylegenerator/plugin/StyleGeneratorPlugin.java
package com.stylegenerator.plugin;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

/**
 * Main plugin class.
 */
public class StyleGeneratorPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {
        try {
            project.getExtensions().create("styleGenerator", StyleGeneratorPluginExtension.class);
            project.getTasks().create("generateStyle", StyleGeneratorPluginTask.class);
        } catch (Exception e) {
            e.printStackTrace();
            throw e;
        }
    }
}

File: plugin/src/main/java/com/stylegenerator/plugin/StyleGeneratorPlugin.kt
package com.stylegenerator.plugin

import org.gradle.api.Plugin
import org.gradle.api.Project

/**
 * Main plugin class.
 */
class StyleGeneratorPlugin : Plugin<Project> {

    override fun apply(project: Project) {

        try {
            project.extensions.create("styleGenerator", StyleGeneratorPluginExtension::class.java)
            val generateStyleTask = project.tasks.create("generateStyle", StyleGeneratorPluginTask::class.java)

            val cleanTask = project.tasks.getByName("clean")
            generateStyleTask.dependsOn(cleanTask)
        } catch (e: Exception) {
            e.printStackTrace()
            throw e
        }
    }
}
